src/experiments/netChange/api.js

```diff
@@ -31,17 +31,16 @@ var netChange = class netChange extends ExtensionAPI {
             name: "netChange.onConnectionChanged",
             register: fire => {
               let observer = async (subject, topic, data) => {
-                // if we get "up" event we should fire an event.
-                if (data === "up") {
-                  last_event = Date.now();
-                  fire.async(data);
-                }
-
-                if (data === "changed") {
-                  // We will coalesce event that are less than 30s apart.
-                  if ( Date.now() - last_event > 30000 &&  gNetworkLinkService.linkStatusKnown && gNetworkLinkService.isLinkUp) {
-                    last_event = Date.now();
-                    fire.async(data);
+                if (netChangeWaiting == false) {
```

Comment:
Not sure on JS code style, but returning early
```
if (netChangeWaiting) {
   return;
}
```
seems much nicer. Same for data != changed/up
Not a big deal though.